README.md

```diff
@@ -109,20 +109,21 @@ You can also launch multiple multi-UA scenarios concurrently using
 non-blocking mode:
 
 ```python
-scens = []
+finalizers = []
 for path, scen in pysipp.walk('path/to/scendirs/root/', proxyaddr=('10.10.8.1', 5060)):
     print("Running scenario collected from {}".format(path))
     scen(block=False)
-    scens.append(scen)
+    finalizers.append(scen)
 
 # All scenarios should now be running concurrently so we can continue
 # program execution...
 print("Continuing program execution...")
 
-# Wait to collect all the results
+# Calling `scen` returns a finalizer that can be
+# used to wait for each scenario with a timeout
 print("Finalizing all scenarios and collecting results")
-for scen in scens:
-  scen.finalize()
+for finalizer in finalizers:
+  finalizer(timeout=10)
```

Comment:
I think we've talked about it in passing but, we might want to move to an explicit method name here. I'm not against `.__call__()` per say but it does seem a bit too implicit the more we've let this sit.

Reply:
hello, are you talking about the ScenarioType having both `run` and `__call__` methods?
the finalizer returned is just a function. Docs are incorrect here thought i will fix them, calling `scen` returns the finalizer

Reply:
I'm talking about 2 things:
- `Scenario.run()` I think should be the api we encourage using.
- you dropping `.finalize()` as part of the runner api.

Right now you've made things arguably more indirected by returning a closure from `Scenario.__call__()` **only when** `block=False`. I wouldn't call this any more *functional* per say other then that you return a closure instead of the runner.

Reply:
The user should have access to the finalizer only when `block=False`. It would make much more sense if the finalizer was a method on the `Scenario` instead of juggling between objects, but that wasn't the case in the existing implementation.

Reply:
> It would make much more sense if the finalizer was a method on the Scenario instead of juggling between objects, but that wasn't the case in the existing implementation.

I like this idea better then returning an anon closure personally.
Is there any reason we can't add this?

It also would tie multi-script-execution lifetime to the scenario object, and for example if one wants to get the results from the *last* scenario run you just call `Scenario.results()` or `Scenario.finalize()` or wtv we decide. This makes the scen a little more *future-like* which it kind of *is* given we have *non-blocking run* support?

Reply:
This solution will change the existing API and be backwards incompatible btw.

We are trying to implement a sync interface to an async functionality. 

We can make the Scenario an awaitable to simplify the interface, but this would require the user to run async tests. Or the Scenario can just be a context manager.

Reply:
> This solution will change the existing API and be backwards incompatible btw.

Not if we make `Scenario.__call__()` a thing tho right?

> We are trying to implement a sync interface to an async functionality.

Yes which is why this needs thorough thought and ideally more then just our opinions 😂 

>  Scenario can just be a context manager.

This idea i don't mind. If we're going to offer a sync interface it may also make sense to consider spawning trio in another thread and then having the main thread make calls to it with `trio.to_thread()` and friends.

We should eventually be supporting async code for spawning and scenarios especially to support multi-script cases where a user wants to do cmd restarts, cancels, etc.